packages/gatsby-plugin-utils/src/polyfill-remote-file/utils/get-request-headers-for-url.ts

```diff
@@ -0,0 +1,27 @@
+import url from "url"
+import path from "path"
+import importFrom from "import-from"
+import resolveFrom from "resolve-from"
+
+import type { Headers } from "got"
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function getStore(): any {
+  const gatsbyPkgRoot = path.dirname(
+    resolveFrom(process.cwd(), `gatsby/package.json`)
+  )
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const { store } = importFrom(gatsbyPkgRoot, `gatsby/dist/redux`) as any
```

Comment:
When I try to use the actual type instead of any it seems to cause problems with type resolution elsewhere. I assume this is some kind of nested dep recursion thing? @wardpeet do you know how to import the type without causing those kinds of problems?

Reply:
Or should I move this util into `gatsby` and import it into `gatsby-plugin-utils`? 🤔

Reply:
Hmm, I did that in https://github.com/gatsbyjs/gatsby/pull/35655/commits/4ba7177b8351ad13d7aabd6d5effabb5aa714415 but now bootstrap is failing with the following:

```
src/polyfill-remote-file/http-routes.ts(8,41): error TS2307: Cannot find module 'gatsby/dist/utils/get-request-headers-for-url' or its corresponding type declarations.
src/polyfill-remote-file/jobs/gatsby-worker.ts(6,41): error TS2307: Cannot find module 'gatsby/dist/utils/get-request-headers-for-url' or its corresponding type declarations.
src/polyfill-remote-file/placeholder-handler.ts(10,41): error TS2307: Cannot find module 'gatsby/dist/utils/get-request-headers-for-url' or its corresponding type declarations.
src/polyfill-remote-file/transform-images.ts(7,41): error TS2307: Cannot find module 'gatsby/dist/utils/get-request-headers-for-url' or its corresponding type declarations.
```

Not entirely sure how to make it work 😅 it does work locally for some reason